web-search-engines/langchain4j-web-search-engine-searchapi/src/main/java/dev/langchain4j/web/search/searchapi/SearchApiWebSearchEngine.java

```diff
@@ -0,0 +1,171 @@
+package dev.langchain4j.web.search.searchapi;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.Utils.isNotNullOrBlank;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+import static java.time.Duration.ofSeconds;
+import static java.util.stream.Collectors.toList;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import dev.langchain4j.web.search.WebSearchEngine;
+import dev.langchain4j.web.search.WebSearchInformationResult;
+import dev.langchain4j.web.search.WebSearchOrganicResult;
+import dev.langchain4j.web.search.WebSearchRequest;
+import dev.langchain4j.web.search.WebSearchResults;
+import dev.langchain4j.web.search.searchapi.result.OrganicResult;
+import lombok.Builder;
+
+/**
+ * Represents SearchApi Search API as a {@code WebSearchEngine}.
+ * See more details <a href="https://www.searchapi.io/docs/google">here</a>.
+ */
+public class SearchApiWebSearchEngine implements WebSearchEngine {
+	public static final String DEFAULT_ENV_VAR = "SEARCHAPI_API_KEY";
+    public static final String DEFAULT_BASE_URL = "https://www.searchapi.io";
+    public static final String DEFAULT_ENGINE = "google";
+    private static final Integer DEFAULT_MAX_RESULTS = 5;
+
+    private final String apiKey;
```

Comment:
`apiKey`, `engine` and `logRequests` don't have to be instance variables

Reply:
Yes, they don't have to be but there's nothing wrong if they are left as-is.